base/tps/src/main/java/org/dogtagpki/server/tps/rest/base/ProfileProcessor.java

```diff
@@ -0,0 +1,495 @@
+                properties.put("Status", status);
+            }
+            auditTPSProfileChange(principal, ILogger.SUCCESS, method, id, properties, null);
+
+            return profileData;
+
+        } catch (PKIException e) {
+            logger.error(method + e.getMessage(), e);
+            auditTPSProfileChange(principal, ILogger.FAILURE, method, id, null, e.toString());
+            throw e;
+
+        } catch (Exception e) {
+            logger.error(method + e.getMessage(), e);
+            auditTPSProfileChange(principal, ILogger.FAILURE, method, id, null, e.toString());
+            throw new PKIException(e);
+        }
+    }
+
+    public ProfileData getProfile(List<String> authorizedProfiles, String profileID) {
+        String method = "ProfileProcessor.getProfile:";
+        String msg = "";
+        logger.info("{} Retrieving profile {}", method, profileID);
+
+        if (profileID == null) {
+            throw new BadRequestException(method + "Missing profile ID");
+        }
+        try {
+           if ((authorizedProfiles== null) || (!authorizedProfiles.contains(UserResource.ALL_PROFILES) && !authorizedProfiles.contains(profileID))) {
+                msg = "profile record restricted for profileID:" + profileID;
+                logger.debug("{} {}", method, msg);
+                throw new PKIException(msg);
+            }
+            ProfileRecord profileRecord = database.getRecord(profileID);
+            return createProfileData(profileRecord);
+
+        } catch (PKIException e) {
+            logger.error(method + e.getMessage(), e);
+            throw e;
+
+        } catch (Exception e) {
+            logger.error(method + e.getMessage(), e);
+            throw new PKIException(e);
+        }
+    }
+
+    public ProfileData updateProfile(Principal principal, List<String> authorizedProfiles, String profileID, ProfileData profileData) {
+        String method = "ProfileProcessor.updateProfile:";
+        String msg = "";
+
+        if (profileID == null) {
+            auditConfigTokenGeneral(principal, ILogger.FAILURE, method, null, "Missing profile ID");
+            throw new BadRequestException("Missing profile ID");
+        }
+        logger.info("{} Updating profile {}", method, profileID);
+
+        if (profileData == null) {
+            auditConfigTokenGeneral(principal, ILogger.FAILURE, method, null, "Missing profile data");
+            throw new BadRequestException("Missing profile data");
+        }
+
+        Map<String, String> properties = profileData.getProperties();
+        for (String name : properties.keySet()) {
+            if (!PROPERTY_NAME_PATTERN.matcher(name).matches()) {
+                throw new BadRequestException("Invalid profile property: " + name);
+            }
+        }
+        try {
+            if ((authorizedProfiles== null) || ((authorizedProfiles != null) && !authorizedProfiles.contains(UserResource.ALL_PROFILES) && !authorizedProfiles.contains(profileID))) {
```

Comment:
The `authorizedProfiles != null` probably can be removed too like in `getProfile()`.

Reply:
Updated